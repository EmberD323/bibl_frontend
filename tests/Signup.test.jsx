

import { describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Signup from "../src/components/Users/Signup.jsx";

describe("Signup component", () => {

  it("renders as per snapshot", () => {
    const { container } = render(<Signup />);
    expect(container).toMatchSnapshot();
  });
  //form errors
  it("renders form error if input is blank", async () => {
    render(<Signup />);
    const user = userEvent.setup();
    const button = screen.getByRole("button");
    await user.click(button);
    await waitFor(() => {
      const firstNameError = screen.getByText('Please fill out this field.', { selector: 'span' });
      expect(firstNameError).toBeInTheDocument();
    });
  });
  // it("on submit, name contains numbers. Error shown", async () => {
   
  // });
  // it("on submit, name too long. Error shown", async () => {
   
  // });
  // it("on submit, email already used. Error shown", async () => {
   
  // });
  // it("on submit, passwords atleast 8 characters. Error shown", async () => {
   
  // });
  // it("on submit, passwords doesnt include numbers. Error shown", async () => {
   
  // });
  // it("on submit, passwords dont match. Error shown", async () => {
   
  // });
  // it("navigates to login on successful submit", async () => {
   
  // });

  // it("renders as not clicked after button click with error", async () => {
  //   const user = userEvent.setup();
  //   render(<Signup />);
  //   const button = screen.getByRole("button");
  //   await user.click(button);
  //   expect(screen.getByTestId('custom-element').textContent).toMatch(/Clicked/i);
  // });


});
