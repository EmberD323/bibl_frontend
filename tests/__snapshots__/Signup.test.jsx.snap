// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Signup component > renders as per snapshot 1`] = `
<div>
  <div
    class="signUp"
  >
    <h2>
      Sign up
    </h2>
    <form>
      <div
        class="firstName"
      >
        <label
          for="first_name"
        >
          First Name
        </label>
        <input
          id="first_name"
          name="first_name"
          required=""
          type="text"
          value=""
        />
      </div>
      <div
        class="lastName"
      >
        <label
          for="last_name"
        >
          Last Name
        </label>
        <input
          id="last_name"
          name="last_name"
          required=""
          type="text"
          value=""
        />
      </div>
      <div
        class="username"
      >
        <label
          for="username"
        >
          Email
        </label>
        <input
          id="username"
          name="username"
          required=""
          type="email"
          value=""
        />
      </div>
      <div
        class="password"
      >
        <label
          for="password"
        >
          Password
        </label>
        <input
          id="password"
          name="password"
          required=""
          type="password"
          value=""
        />
      </div>
      <div
        class="passportConfirm"
      >
        <label
          for="passwordConfirm"
        >
          Password
        </label>
        <input
          id="passwordConfirm"
          name="passwordConfirm"
          required=""
          type="password"
          value=""
        />
      </div>
      <button
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</div>
`;
